### ISSUE TITLE
Route parameters not generated

### ISSUE THREAD

**Issue by guplem** [Milad-Akarie/auto_route_library#1508]
I am trying to navigate to a page/route (`RejectionRoute`) but the parameters of the widget seem not to be created by the `auto_route_generator`.

> This is my environment:
> - Flutter (Channel stable, 3.7.10, on Microsoft Windows [Version 10.0.22621.1555], locale en-150)
> - auto_route_generator: 6.2.0
> - auto_route: 6.3.0

This is how I try to navigate to the page:
```Dart
context.router.push(RejectionRoute(thing: thingsFS, thingId: thingsFS.id, groupId: thingsFS.groupId))
```

But I get an error with the `thingId` and `groupId` parameters: `The named parameter 'thingId' isn't defined.` and `The named parameter 'groupId' isn't defined.`.

Both of those parameters are defined in the widget.
This is the Stateful widget defining the page (without the state): 
```Dart
@RoutePage()
class RejectionScreen extends StatefulWidget {
  final ThingFS thing;
  final String? thingId;
  final String groupId;

  const RejectionScreen({Key? key, @PathParam.inherit("id") required this.thingId, @PathParam.inherit("group") required this.groupId, required this.thing}) : super(key: key);

  @override
  State<RejectionScreen> createState() => _RejectionScreenState();
}
```

As you can see, both of the parameters that cause the issue are inherited. Maybe the issue is related to that? The `thing` parameter seems to work fine.

I have the following app_router.dart:
```Dart
import 'package:auto_route/auto_route.dart';
import 'app_router.gr.dart';

@AutoRouterConfig(replaceInRouteName: 'Screen,Route')
class AppRouter extends $AppRouter {
  @override
  RouteType get defaultRouteType => const RouteType.material(); //.cupertino, .adaptive ..etc

  @override
  List<AutoRoute> get routes => [
    AutoRoute(
      page: ThingDetailsRoute.page,
      path: "/thing/:group/:id",
      children: [
        AutoRoute(page: RejectionRoute.page, path: 'rejection'),
      ],
    ),
    AutoRoute(page: SettingsRoute.page, path: "/settings"),
    AutoRoute(page: AuthRoute.page, path: "/"),
  ];
}
```

The navigation to the `ThingDetailsRoute` (the page from which the parameters are inherited) works flawlessly, like this: 
```Dart
context.router.push(ThingDetailsRoute(thing: thingsFS, thingId: thingsFS.id, groupId: thingsFS.groupId))
```

Any clue on what I might have done wrong? 

I did have it working without inheriting the parameters, but I thought having the inherited would be a better practice. But after inheriting them, it looks like the generation deleted those parameters:
![image](https://user-images.githubusercontent.com/11029629/233827223-511d3078-2748-4055-b9c2-91cf1276e1aa.png)

**Comment by guplem** [comment]
I could work around the issue by stopping using the inherited parameters/child routes.

I did change the constructor of the screen to:
```Dart
const RejectionScreen({Key? key, @PathParam("id") required this.thingId, @PathParam("group") required this.groupId, required this.thing}) : super(key: key);
```

While changing the app_router.dart to:
```Dart
import 'package:auto_route/auto_route.dart';
import 'app_router.gr.dart';

@AutoRouterConfig(replaceInRouteName: 'Screen,Route')
class AppRouter extends $AppRouter {
  @override
  RouteType get defaultRouteType => const RouteType.material(); //.cupertino, .adaptive ..etc

  @override
  List<AutoRoute> get routes => [
    AutoRoute(
      page: ThingDetailsRoute.page,
      path: "/thing/:group/:id",
    ),
    AutoRoute(page: RejectionRoute.page, path: "/thing/rejection/:group/:id"),
    AutoRoute(page: SettingsRoute.page, path: "/settings"),
    AutoRoute(page: AuthRoute.page, path: "/"),
  ];
}
```

But I believe this is not ideal and the inheritance should be used instead. 
Still haven't figured out why my original approach doesn't work.

**Comment by guplem** [comment]
Also, I have noticed that [the example in the documentation about inherited parameters](https://pub.dev/packages/auto_route#:~:text=Inherited%20Path%20Parameters) is using the widget name instead of the Route's page:
![image](https://user-images.githubusercontent.com/11029629/233906912-d27d8fbb-35de-46db-8d91-78b6e17c20ac.png)

However, If I try to use it, I get the following error:
```
The argument type 'Type' can't be assigned to the parameter type 'PageInfo<dynamic>'.
```
So, I have to use `ThingDetailsRoute.page` instead of `ThingDetailsScreen` for the `page`parameter of the AutoRoute factory. Is the documentation outdated or am I doing something wrong?

**Comment by guplem** [comment]
I just realized of the following warning while generating the routes with the following command `flutter pub run build_runner build --delete-conflicting-outputs`:
```
[WARNING] auto_route_generator on lib/feature/rejection/rejection_screen.dart:

WARNING => Because [RejectionScreen] has required parameters (thing) that can not be parsed from path,
@PathParam() and @QueryParam() annotations will be ignored.
``` 

However, that warning is not always present, and it was only when I was going back and forth commits that I could see it. 
I believe it is because if there are no changes made into the code, no generation must be done.
![image](https://user-images.githubusercontent.com/11029629/234467588-85b4cc3d-4aeb-49ea-97ed-5c210a9e7a31.png)

But anyway, even when the warning is present... it doesn't look like a warning at all, the `Succeeded` at the end makes it look like there were no issues at all, but the flutter developer made a clear mistake that **must** be fixed.

I don't think it is a common issue since I couldn't find much information online, but it took me multiple days to be able to address it.
Even now that I know that the warning should be present, I am struggling to print it again on the console again. I am not even sure about how I managed to see it **once**.

I think the warnings of the `auto_route_generator` should be improved.

The bottom line, **How to solve the issue**:
Remove the `required` flag from the parameters of the screen/widget referenced by the route. Then regenerate the routes.

**Comment by Milad-Akarie** [comment]
Hey @guplem sorry for the late response, I just finished my vacation, What's happening is actually indented, inherited path params are only obtained from the parent path, we should not pass them manually

**Comment by guplem** [comment]
@Milad-Akarie Hope you had a great vacation!

During the fixing of the issue, I got a question and a complain:



### When to use @PathParam.inherit?

> I know your comment is related to this, but I didn't understand what you mean by "*we should not pass them manually*"

I did remove the `.inherit` from `@PathParam.inherit` in order to have the error go away. Now the constructor looks like this:
```Dart
const RejectionScreen({@PathParam("id") required this.thingId, @PathParam("group") required this.groupId, this.thing, Key? key}) : super(key: key);
``` 
And the route is defined like this:
```Dart
AutoRoute(
      page: ThingDetailsRoute.page,
      path: "/thing/:group/:id",
      children: [
        AutoRoute(page: RejectionRoute.page, path: 'rejection'),
      ],
    ),
```
Could you clarify when should the `@PathParam.inherit` be used and if my solution is correct? In [this issue](https://github.com/Milad-Akarie/auto_route_library/issues/1443#issuecomment-1489175278) you suggest that it should be used, and [in the documentation](https://pub.dev/packages/auto_route#:~:text=Inherited%20Path%20Parameters) it looks like it should be used too. So, I am confused.



### Warnings of route generation are not clearly visible

I saw the following warning while generating the routes with the following command `flutter pub run build_runner build --delete-conflicting-outputs`:
```
[WARNING] auto_route_generator on lib/feature/rejection/rejection_screen.dart:

WARNING => Because [RejectionScreen] has required parameters (thing) that can not be parsed from path,
@PathParam() and @QueryParam() annotations will be ignored.
``` 

However, that warning was **not** always present, and it was only when I was going back and forth commits that I could see it. 

But, even now that I know that the warning should be present, I am struggling to print it on the console again. I am not even sure about how I managed to see it **once**.

And, even when the warning is present... it doesn't look like a warning at all, the `Succeeded` at the end makes it look like there were no issues at all, but the flutter developer made a clear mistake that **must** be fixed.

I think these warnings of the `auto_route_generator` should be improved.

**Comment by guplem** [comment]
@Milad-Akarie I would appreciate it if you could clarify my doubts, maybe it is useful information/documentation of the package's usage

**Comment by tomasweigenast** [comment]
I think he will not answer
